src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,294 @@
+package duke;
+
+import duke.data.exception.DukeException;
+import duke.data.task.Deadline;
+import duke.data.task.Event;
+import duke.data.task.Task;
+import duke.data.task.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Duke {
+
+    private static final String FOLDER_PATH = "data";
+    private static final String FILE_NAME = "duke.txt";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String LIST = "list";
+    private static final String DONE = "done";
+    private static final String BYE = "bye";
+    private static final String BY = "/by";
+    private static final String AT = "/at";
+    private static final String DELETE = "delete";
+    private static final String DOTTED_LINE = "____________________________________________________________";
+    private static final String LOGO = (" ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n")
+            .replaceAll("\n", System.lineSeparator() + "\t");
+
+    private static final int INSTRUCTION_LENGTH = 2;
+
+    private final ArrayList<Task> tasks;
+    private int taskCount;
+
+    private boolean isLoaded;
+
+    Duke() {
+        tasks = new ArrayList<>();
+        taskCount = 0;
+        isLoaded = false;
+    }
+
+    public static void main(String[] args) {
+        Duke duke = new Duke();
+        duke.loadTasks();
+        duke.greetUser();
+        duke.processInput();
+    }
+
+    private void printWithIndent(String string) {
+        System.out.println("\t" + string);
+    }
+
+    private void loadTasks() {
```

Comment:
Method is longer than 30 lines of code. Could consider extracting some code fragments to a method